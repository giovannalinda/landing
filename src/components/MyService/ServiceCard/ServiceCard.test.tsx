import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { FiCode } from 'react-icons/fi'
import { ServiceCard } from '.'

describe('ServiceCard', () => {
  it('should render correctly', async () => {
    render(
      <ServiceCard title='Front End Development'>
        <FiCode aria-label='code icon' size={40} strokeWidth={1} />
      </ServiceCard>,
    )

    expect(
      await screen.findByText(/Front End Development/i),
    ).toBeInTheDocument()
    expect(await screen.findByLabelText(/code icon/i)).toBeInTheDocument()
  })

  it('should match snapshot', async () => {
    const { container } = render(
      <ServiceCard title='Front End Development'>
        <FiCode size={40} strokeWidth={1} />
      </ServiceCard>,
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it("should render 'discuss now' when user hover button", async () => {
    render(
      <ServiceCard title='Front End Development'>
        <FiCode aria-label='code icon' size={40} strokeWidth={1} />
      </ServiceCard>,
    )

    userEvent.hover(
      await screen.findByRole('button', {
        name: /Front End Development/i,
      }),
    )

    expect(await screen.findByText(/Discuss now/i)).toBeInTheDocument()
  })
})
