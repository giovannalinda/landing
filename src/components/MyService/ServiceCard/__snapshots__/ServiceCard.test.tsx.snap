// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCard should match snapshot 1`] = `
<button
  aria-label="Front End Development"
  class="sc-kgUAyh gAHRgT"
>
  <svg
    fill="none"
    height="40"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1"
    viewBox="0 0 24 24"
    width="40"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polyline
      points="16 18 22 12 16 6"
    />
    <polyline
      points="8 6 2 12 8 18"
    />
  </svg>
  <div
    class="sc-hTtwUo kzOLOS"
  >
    Front End Development
    <hr
      class="sc-jgbSNz kVUrhz"
    />
    <div
      class="sc-gsnTZi freJdh"
    >
      <span>
        Discuss now
      </span>
      <svg
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M23.0677 11.9929L18.818 7.75739L17.4061 9.17398L19.2415 11.0032L0.932469 11.0012L0.932251 13.0012L19.2369 13.0032L17.4155 14.8308L18.8321 16.2426L23.0677 11.9929Z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</button>
`;
